File: app/src/main/java/com/aria/reply/AriaTestReceiver.java
package com.aria.reply;
import android.content.*;
import android.app.*;
import android.os.Bundle;
public class AriaTestReceiver extends BroadcastReceiver{
 public static final String ACTION="com.aria.reply.SYNTHETIC";
 @Override public void onReceive(Context c,Intent i){
   AriaStore s=new AriaStore(c);
   s.addMessage("Synthetic Contact","user","Bhai mujhe bhej de");
   s.put("last_capture","Synthetic Contact: Bhai mujhe bhej de");
 }
}

File: aria-reply-native/app/src/main/java/com/aria/reply/AriaTestReceiver.kt
package com.aria.reply

import android.app.NotificationManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.app.RemoteInput

class AriaTestReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val results: Bundle? = RemoteInput.getResultsFromIntent(intent)
        val reply = results?.getCharSequence(AriaTestNotification.RESULT_KEY)?.toString().orEmpty()
        val store = AriaStore(context)
        if (reply.isNotBlank()) {
            store.lastReply = reply
            store.lastError = ""
            store.logEvent("Synthetic RemoteInput reply received by Aria test receiver", true)
        } else {
            store.lastError = "Synthetic RemoteInput returned no reply"
            store.logEvent("Synthetic RemoteInput receiver got no text", false)
        }
        context.getSystemService(NotificationManager::class.java)?.cancel(AriaTestNotification.NOTIFICATION_ID)
    }
}
